model/operation.cc

```diff
@@ -435,9 +435,32 @@ namespace minsky
     switch (classify(type()))
       {
       case function: case reduction: case scan: case tensor:
-        if (check && !ports[1]->units(check).empty())
-          throw_error("function input not dimensionless");
-        return {};
+        switch (type())
+          {
+		  case percent:
+		    {
+             if (check && !ports[1]->units(check).empty())
+               throw_error("function input not dimensionless");
```

Comment:
For percent, we can have arbitrary units, ISTM. 10% of a metre is a decimetre, for instance. Although the most obvious implementation would be to return units %m in that case, which would look a bit ugly. Similarly %² would refer to multiplying by 10,000.

Reply:
Thanks. If I undestand correctly, this means I should get rid of this if conditional. The function input does not have to be unitless.